### PR TITLE
release: v1.24.0

### PR DESCRIPTION
**Pull request by jennijuju**
## Related Issues

need cherry pick

https://github.com/filecoin-project/lotus/pull/11440
https://github.com/filecoin-project/lotus/pull/11437
https://github.com/filecoin-project/lotus/pull/11175

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,85 +14,84 @@ The Filecoin network version 21 delivers the following FIPs:
...
+Note to full archival node operators: you may expect it takes some time  for the node to complete the final migration, during this period your node will fall out of sync and your chain service may have some disruption. However, you can expect the node to catch up soon after the migration completes. You can test out the migration by running the following on your node in offline mode:
 
-```
-lotus state actor-cids --network-version=21
-Network Version: 21
-Actor Version: 12
-Manifest CID: bafy2bzacednzb3pkrfnbfhmoqtb3bc6dgvxszpqklf3qcc7qzcage4ewzxsca
+1. `lotus chain head | head -n1`
+2. Stop Lotus daemon
+3. `./lotus-shed migrate-state --repo=[path-to-your-lotus-repo] 22 [output-of-step-1]`
```
Comment (rjan90) [inline CHANGELOG.md:65]: The 22 here should be 21.

We can also add a link to the Lotus docs for testing this: https://lotus.filecoin.io/kb/test-migration/

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,85 +14,84 @@ The Filecoin network version 21 delivers the following FIPs:
...
 
------------------
+## Migration  
 
+We are expecting a heavier than nommal state migration for this upgrade due to the amount of the state changes introduced for miner sector info. (This a similar migration node as the Shark upgrade, however, we have introduced a couple migration performance optimization since then for a smoother upgrade.)
 
+All node operators, including storage providers, should be aware that ONE pre-migration is being scheduled 180 epochs before the upgrade, around 2023-12-12T12:00:00Z. It will take around 20-30min for the migration and less than 30sec for the final migration, depending on the amount of the historical state in the node blockstore and the hardwaree specs the node is running on. During this time, expect slower block validation times, increased CPU and memory usage, and longer delays for API queries (during our testing, it topped out about 205 RAM(htop) on a 1TiB box).
 
-## v12 Builtin Actor Bundles
+We recommend node operators (who haven't enbabled splistore `universal` mode) that do not care about historical chain states, to prune the chain blockstore by syncing from a snapshot 1-2 days before the upgrade.
```
Comment (rjan90) [inline CHANGELOG.md:59]:
```suggestion
We recommend node operators (who haven't enabled splitstore `discard` mode) that do not care about historical chain states, to prune the chain blockstore by syncing from a snapshot 1-2 days before the upgrade.
```

File: CHANGELOG.md (hunk 1 above)
Comment (rjan90) [inline CHANGELOG.md:65]:
```suggestion
1. `lotus chain head | head -n1`
2. Stop Lotus daemon
3. `./lotus-shed migrate-state --repo=[path-to-your-lotus-repo] 21 [output-of-step-1]`

You can check out the [tutorial for benchmarking the network migration here.](https://lotus.filecoin.io/kb/test-migration/)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -20,85 +14,84 @@ The Filecoin network version 21 delivers the following FIPs:
...
-- chore: deps: update to go-state-types v0.12.5 ([filecoin-project/lotus#11339](https://github.com/filecoin-project/lotus/pull/11339))
-- chore: backport calibnet lightweight patch ([filecoin-project/lotus#11423](https://github.com/filecoin-project/lotus/pull/11423))
+- chore: deps: update to final actors ([filecoin-project/lotus#11330](https://github.com/filecoin-project/lotus/pull/11440))
+- chore: deps: update to go-state-types v0.12.8 ([filecoin-project/lotus#11339](https://github.com/filecoin-project/lotus/pull/11437))
+- chore: deps: update libp2p to v0.30.0 #11434
+
+
+## Snapshots 
+
+The [Forest team](https://filecoinproject.slack.com/archives/C029LPZ5N73) at Chainsafe has launched a brand new lightweight snapshot service that is backed up by forest nodes! This is a great alternative service along with the fil-infra one, and it is compatiable with lotus! We recommend lotus users to check it out [here](https://docs.filecoin.io/networks/mainnet#resources)!
```
Comment (rjan90) [inline CHANGELOG.md:94]:
```suggestion
The [Forest team](https://filecoinproject.slack.com/archives/C029LPZ5N73) at Chainsafe has launched a brand new lightweight snapshot service that is backed up by forest nodes! This is a great alternative service along with the fil-infra one, and it is compatible with lotus! We recommend lotus users to check it out [here](https://docs.filecoin.io/networks/mainnet#resources)! 
```

**Review by rjan90 — APPROVED** [review]
A couple of nits, but else looks good!

**Review by aarshkshah1992 — APPROVED** [review]
LGTM.

**Pull request merged**